Sources/AsyncStream.swift

```diff
@@ -1,18 +1,24 @@
 public protocol AsyncSending {
-    func send(_ data: Data, timingOut deadline: Double, completion: (Void throws -> Void) -> Void)
+    associatedtype Sent
+    func send(_ data: Sent, timingOut deadline: Double, completion: (Void throws -> Void) -> Void)
     func flush(timingOut deadline: Double, completion: (Void throws -> Void) -> Void)
 }
 
 public protocol AsyncReceiving {
-    func receive(upTo byteCount: Int, timingOut deadline: Double, completion: (Void throws -> Data) -> Void)
+    associatedtype Received
+    func receive(upTo byteCount: Int, timingOut deadline: Double, completion: (Void throws -> Received) -> Void)
 }
 
 public protocol AsyncSendingStream: Closable, AsyncSending {}
 public protocol AsyncReceivingStream: Closable, AsyncReceiving {}
-public protocol AsyncStream: AsyncSendingStream, AsyncReceivingStream {}
+public protocol AsyncStream: AsyncSendingStream, AsyncReceivingStream {
+    associatedtype Interchanged
```

Comment:
My vote is for `TransferType` here.  I think having types as past tense verbs should be avoided.